### ISSUE TITLE
Uploading DIF from Windows fails with "Invalid checksum" eror

### ISSUE THREAD

**Issue by petrmanek** [getsentry/sentry-cli#1906]
### Environment

- Windows 10, x86_64
- sentry-cli v2.25.3

### Steps to Reproduce

1. Open Visual Studio 2019 Developer Command Prompt
2. Build a C++ project using cmake with `CMAKE_BUILD_TYPE` set to `RelWithDebInfo`
3. Run: `sentry-cli upload-dif -o %SENTRY_ORGANIZATION% -p %SENTRY_PROJECT% --log-level=debug .`

### Expected Result

All debug information files should be uploaded to Sentry.

### Actual Result

The sentry-cli call fails with exit code 1. The following message is printed:

```error: Invalid checksum```

This runs on a Windows VM. Please note that I run similar jobs on macOS and Linux VMs on the same computing cluster, which also call `sentry-cli upload-dif` with similar inputs at some point. Neither of these ever fail because of Sentry, only the Windows job does.

### Logs

Here are (sanitized) logs from a typical run: [sentry_fail.log](https://github.com/getsentry/sentry-cli/files/14001660/sentry_fail.log)

**Comment by szokeasaurusrex** [comment]
Hi @petrmanek, thank you for reporting this issue. Have you ever been able to run this command successfully on Windows? If yes, what was the most recent version of the Sentry CLI where running the command succeeded?

**Comment by petrmanek** [comment]
> Have you ever been able to run this command successfully on Windows?

Yes, in fact I was running this in my CI/CD pipeline as recently as 2 weeks ago. The only thing that changed was time and some features of my software. I recently switched my C++ standard from C++17 to C++20. Could that have anything to do with it?

> If yes, what was the most recent version of the Sentry CLI where running the command succeeded?

Everything worked with v2.12.0 until about 2 weeks ago, when the error started appearing. Since then I tried updating to the latest version (v2.25.3 as if writing), which made no difference.

**Comment by szokeasaurusrex** [comment]
> I recently switched my C++ standard from C++17 to C++20. Could that have anything to do with it?

Perhaps, if making that change could have changed the debug information file format. You can use the [`sentry-cli debug-files check`](https://docs.sentry.io/product/cli/dif/#checking-files) command to check whether the debug information files are usable. More information about the debug information file formats we accept is available [here](https://docs.sentry.io/platforms/native/data-management/debug-files/?original_referrer=https%3A%2F%2Fwww.google.com%2F).

Please try running the `sentry-cli debug-files check` command against the files which are causing the issue, and let me know what the output is.

**Comment by petrmanek** [comment]
> Perhaps, if making that change could have changed the debug information file format

I will try that. A job is currently running so I should have results in approximately an hour.

In the meantime, looking at logs I do not see anything out of ordinary. Mostly DLLs (x86_64 library) and PDBs (x86_64 debug companion) and a single EXE (x86_64 executable). Correct me if I am wrong, but if sentry-cli identifies these files correctly and displays `UPLOADED` next to their name, it implies that they are considered useful, right?

**Comment by petrmanek** [comment]
I have a follow up: `sentry dif check` seems to recognize my files just fine even after switching to C++20. The following would be a typical output:

![20240122_16h08m10s_grim](https://github.com/getsentry/sentry-cli/assets/1297382/9962ed1a-cab8-445a-9bbc-159f141887cf)

**Comment by szokeasaurusrex** [comment]
> Correct me if I am wrong, but if sentry-cli identifies these files correctly and displays `UPLOADED` next to their name, it implies that they are considered useful, right?

The code that the `check` command calls to determine whether the file is useful does not appear to get called from the upload code, so it could be possible that a file might not be considered usable even if it can get uploaded properly. Although I am not sure about that, since the code is pretty complex.

Have you checked whether you are able to see the uploaded debug files in Sentry?

**Comment by petrmanek** [comment]
Apologies for a delayed response.

> Have you checked whether you are able to see the uploaded debug files in Sentry?

Yes, they do and there appear to be no processing issues either.

The following log snippet of sentry-cli:

```
[...]
  UPLOADED 3875da5b-901f-4639-8fa2-5d1883c454fe-1 (plugins\app\csvfile.dll; x86_64 library)
  UPLOADED 7d509770-d1b8-4423-bf65-be6d87dc6162-1 (lib\scn.dll; x86_64 library)
  UPLOADED f4f2beaa-2b80-4bed-95e9-cf7b2e5e91c4-1 (share\debug\app\dacscan.pdb; x86_64 debug companion)
  UPLOADED 3ba9b8a0-3a72-4da4-9d78-080af7e842ee-1 (share\debug\app\mmfile.pdb; x86_64 debug companion)
  UPLOADED 179b21bb-a7d4-44a2-a06d-3b9275daf552-1 (share\debug\app\fitpix.pdb; x86_64 debug companion)
  UPLOADED 9bf1a6ce-5f6c-4040-bba8-bd66084e3f9d-1 (bin\crashpad_wer.pdb; x86_64 debug companion)
  UPLOADED 2337ece1-3804-47cb-86cf-377af91bcd00-1 (plugins\app\livehist.dll; x86_64 library)
[...]
```

...nicely corresponds with what I see on sentry.io:

![20240124_11h30m26s_grim](https://github.com/getsentry/sentry-cli/assets/1297382/0f066ab1-beb0-4c44-a57a-0eee89c9bf4b)

**Comment by szokeasaurusrex** [comment]
@petrmanek Just want to make sure I am understanding everything correctly: my understanding is that besides the error message showing up in the logs, everything appears to be working in Sentry as expected? Or have you noticed anything wrong in Sentry (such as an error message stating that debug files are missing when you view an error event)?

**Comment by petrmanek** [comment]
> besides the error message showing up in the logs, everything appears to be working in Sentry as expected?

That is correct, as far as I can tell. DIFs appear to be identified as useful, uploaded and visible online. Only the process for some reason exits with a failing exit code and prints the `error: Invalid checksum` message before dying.

> Or have you noticed anything wrong in Sentry (such as an error message stating that debug files are missing when you view an error event)?

I have been on a lookout for those. Over the last two weeks, I have not noticed any such incident (or anything similar for that matter).

**Comment by szokeasaurusrex** [comment]
@petrmanek That is good to hear, glad that everything appears to work as expected, besides the error message. 

Since our team is currently busy with other projects, and since the only symptom appears to be the error message in the logs, it might take us some time before we are able to fully investigate this issue.

However, if you begin noticing any other problems as a result of this issue (such as DIFs not being uploaded to Sentry, or a missing debug file error), please let us know right away! Naturally, we would prioritize investigating this issue if it begins to negatively impact your experience using Sentry.

**Comment by petrmanek** [comment]
> Since our team is currently busy with other projects, and since the only symptom appears to be the error message in the logs, it might take us some time before we are able to fully investigate this issue.

As you wish. I will modify my CI's deployment script for Windows to allow for non-zero exit code to be returned by sentry-cli at this stage for as long as this issue remains open. Please let me know if you would like to run any tests.


> However, if you begin noticing any other problems as a result of this issue (such as DIFs not being uploaded to Sentry, or a missing debug file error), please let us know right away! We would, of course, prioritize investigating this issue if it begins to negatively impact your experience using Sentry.

I will do that if/when I notice anything serious. Realistically speaking, however, my pipeline has 50+ jobs, so I am unlikely to spot anything unless I start seeing unsymbolicated crashes at sentry.io.

**Comment by szokeasaurusrex** [comment]
Ok thank you, and I appreciate your understanding! Also, if anyone else experiences this issue, please leave a comment, so we can understand how common this problem is.

**Comment by Mordil** [comment]
We just started getting this error on our macOS GitHub Actions runner.

We're using the [Fastlane plugin](https://github.com/getsentry/sentry-fastlane-plugin), with the Sentry CLI installed via Homebrew. (version 2.31.2)

We started getting this error before we updated, as we were on a slightly older version of the CLI.

I think it started after we updated our runner's OS to macOS 14.2.1. Other than that, we haven't had any changes to our system before we started seeing this error.

Other than just changing our script to allow failures from this - is there another remedy? We updated our Fastlane script to the following

```ruby
  sentry_debug_files_upload(
    auth_token: SENTRY_AUTH_TOKEN,
    org_slug: 'dev-peek',
    project_slug: 'peekpro-inhouse',
    wait: false,
    log_level: 'trace'
  )
```

<details>
<summary>CI debug logs</summary>

<pre>
[10:48:49]: ---------------------------------------
[10:48:49]: --- Step: sentry_debug_files_upload ---
[10:48:49]: ---------------------------------------
[10:48:49]: Using sentry-cli 2.31.2
[10:48:49]: Starting sentry-cli...
[10:48:49]: DEBUG   2024-05-28 10:48:49.609613 -07:00 sentry-cli version: 2.31.2, platform: "darwin", architecture: "arm64"
[10:48:49]: INFO    2024-05-28 10:48:49.610240 -07:00 sentry-cli was invoked with the following command line: "/Users/administrator/actions-runner/_work/nomad/nomad/vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-sentry-1.22.1/bin/sentry-cli-Darwin-universal" "debug-files" "upload" "." "--wait"
[10:48:49]: INFO    2024-05-28 10:48:49.610420 -07:00 Issuing a command for Organization: dev-peek Project: peekpro-inhouse
[10:48:49]: DEBUG   2024-05-28 10:48:49.611011 -07:00 request GET https://sentry.io/api/0/organizations/dev-peek/chunk-upload/
[10:48:49]: DEBUG   2024-05-28 10:48:49.611029 -07:00 using token authentication
[10:48:49]: DEBUG   2024-05-28 10:48:49.611045 -07:00 retry number 0, max retries: 0
[10:48:49]: DEBUG   2024-05-28 10:48:49.646666 -07:00 > GET /api/0/organizations/dev-peek/chunk-upload/ HTTP/1.1
[10:48:49]: DEBUG   2024-05-28 10:48:49.646678 -07:00 > Host: sentry.io
[10:48:49]: DEBUG   2024-05-28 10:48:49.646680 -07:00 > Accept: */*
[10:48:49]: DEBUG   2024-05-28 10:48:49.646682 -07:00 > Connection: TE
[10:48:49]: DEBUG   2024-05-28 10:48:49.646684 -07:00 > TE: gzip
[10:48:49]: DEBUG   2024-05-28 10:48:49.646686 -07:00 > User-Agent: sentry-cli/2.31.2
[10:48:49]: DEBUG   2024-05-28 10:48:49.647350 -07:00 > Authorization: ***
[10:48:49]: DEBUG   2024-05-28 10:48:49.784833 -07:00 < HTTP/1.1 200 OK
[10:48:49]: DEBUG   2024-05-28 10:48:49.784851 -07:00 < server: nginx
[10:48:49]: DEBUG   2024-05-28 10:48:49.784860 -07:00 < date: Tue, 28 May 2024 17:48:49 GMT
[10:48:49]: DEBUG   2024-05-28 10:48:49.784868 -07:00 < content-type: application/json
[10:48:49]: DEBUG   2024-05-28 10:48:49.784876 -07:00 < vary: Accept-Encoding,Accept-Language, Cookie
[10:48:49]: DEBUG   2024-05-28 10:48:49.784882 -07:00 < allow: GET, POST, HEAD, OPTIONS
[10:48:49]: DEBUG   2024-05-28 10:48:49.784889 -07:00 < access-control-allow-methods: GET, POST, HEAD, OPTIONS
[10:48:49]: DEBUG   2024-05-28 10:48:49.784901 -07:00 < access-control-allow-headers: X-Sentry-Auth, X-Requested-With, Origin, Accept, Content-Type, Authentication, Authorization, Content-Encoding, sentry-trace, baggage, X-CSRFToken
[10:48:49]: DEBUG   2024-05-28 10:48:49.784912 -07:00 < access-control-expose-headers: X-Sentry-Error, X-Sentry-Direct-Hit, X-Hits, X-Max-Hits, Endpoint, Retry-After, Link
[10:48:49]: DEBUG   2024-05-28 10:48:49.784920 -07:00 < access-control-allow-origin: *
[10:48:49]: DEBUG   2024-05-28 10:48:49.784927 -07:00 < x-sentry-rate-limit-remaining: 39
[10:48:49]: DEBUG   2024-05-28 10:48:49.784933 -07:00 < x-sentry-rate-limit-limit: 40
[10:48:49]: DEBUG   2024-05-28 10:48:49.784941 -07:00 < x-sentry-rate-limit-reset: 1716918530
[10:48:49]: DEBUG   2024-05-28 10:48:49.784948 -07:00 < x-sentry-rate-limit-concurrentremaining: 24
[10:48:49]: DEBUG   2024-05-28 10:48:49.784954 -07:00 < x-sentry-rate-limit-concurrentlimit: 25
[10:48:49]: DEBUG   2024-05-28 10:48:49.784961 -07:00 < content-language: en
[10:48:49]: DEBUG   2024-05-28 10:48:49.784967 -07:00 < x-frame-options: deny
[10:48:49]: DEBUG   2024-05-28 10:48:49.784975 -07:00 < x-content-type-options: nosniff
[10:48:49]: DEBUG   2024-05-28 10:48:49.784981 -07:00 < x-xss-protection: 1; mode=block
[10:48:49]: DEBUG   2024-05-28 10:48:49.784995 -07:00 < content-security-policy: script-src 'self' 'unsafe-inline' 'report-sample' s1.sentry-cdn.com js.sentry-cdn.com browser.sentry-cdn.com statuspage-production.s3.amazonaws.com static.zdassets.com aui-cdn.atlassian.com connect-cdn.atl-paas.net js.stripe.com 'strict-dynamic' cdn.pendo.io data.pendo.io pendo-io-static.storage.googleapis.com pendo-static-5634074999128064.storage.googleapis.com; connect-src 'self' *.algolia.net *.algolianet.com *.algolia.io sentry.io *.sentry.io s1.sentry-cdn.com o1.ingest.sentry.io api2.amplitude.com app.pendo.io data.pendo.io reload.getsentry.net t687h3m0nh65.statuspage.io sentry.zendesk.com ekr.zdassets.com maps.googleapis.com; worker-src blob:; frame-ancestors 'self' *.sentry.io; img-src * blob: data:; style-src * 'unsafe-inline'; base-uri 'none'; frame-src app.pendo.io demo.arcade.software js.stripe.com sentry.io; default-src 'none'; media-src *; object-src 'none'; font-src * data:; report-uri https://o1.ingest.sentry.io/ap
[10:48:49]: DEBUG   2024-05-28 10:48:49.785006 -07:00 < x-envoy-attempt-count: 1
[10:48:49]: DEBUG   2024-05-28 10:48:49.785012 -07:00 < x-envoy-upstream-service-time: 84
[10:48:49]: DEBUG   2024-05-28 10:48:49.785018 -07:00 < x-served-by: getsentry-web-rpc-production-5775747db-fw2hw
[10:48:49]: DEBUG   2024-05-28 10:48:49.785025 -07:00 < x-sentry-proxy-url: http://10.2.0.67:8999/api/0/organizations/dev-peek/chunk-upload/
[10:48:49]: DEBUG   2024-05-28 10:48:49.785032 -07:00 < x-served-by: getsentry-control-web-default-common-production-57c99dccbdwxrss
[10:48:49]: DEBUG   2024-05-28 10:48:49.785039 -07:00 < x-served-by: frontend-default-6ffdc95f4b-5kpw7
[10:48:49]: DEBUG   2024-05-28 10:48:49.785045 -07:00 < strict-transport-security: max-age=31536000; includeSubDomains; preload
[10:48:49]: DEBUG   2024-05-28 10:48:49.785052 -07:00 < via: 1.1 google
[10:48:49]: DEBUG   2024-05-28 10:48:49.785058 -07:00 < Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
[10:48:49]: DEBUG   2024-05-28 10:48:49.785065 -07:00 < Transfer-Encoding: chunked
[10:48:49]: DEBUG   2024-05-28 10:48:49.786121 -07:00 response status: 200
[10:48:49]: DEBUG   2024-05-28 10:48:49.786133 -07:00 body: {"url":"https://us.sentry.io/api/0/organizations/dev-peek/chunk-upload/","chunkSize":8388608,"chunksPerRequest":64,"maxFileSize":2147483648,"maxRequestSize":33554432,"concurrency":8,"hashAlgorithm":"sha1","compression":["gzip"],"accept":["debug_files","release_files","pdbs","sources","bcsymbolmaps","il2cpp","portablepdbs","artifact_bundles","artifact_bundles_v2"]}
[10:48:57]: error: Invalid checksum
[10:48:57]: DEBUG   2024-05-28 10:48:57.578688 -07:00 skipping update nagger because session is not attended
</pre>
</details>

**Comment by petrmanek** [comment]
Just to update this: we are still seeing this on Windows, there has been no change since the issue was reported half a year ago.

**Comment by szokeasaurusrex** [comment]
@Mordil Please open a [separate issue](https://github.com/getsentry/sentry-cli/issues/new?assignees=&labels=&projects=&template=BUG_REPORT.yml), since the problem you are describing seems to be different from the problem @petrmanek is experiencing.

@petrmanek only observed the problem on Windows and besides the error showing in the logs, everything in Sentry works properly.

On the other hand, from what you have written @Mordil, it sounds like you are noticing something wrong in Sentry, since you don't see releases and debug info in Sentry. You are also observing this problem on Mac, rather than Windows.

Please provide as much detail as possible when opening your issue; this will help me to reproduce your issue and to identify a fix/workaround.

**Comment by Mordil** [comment]
@szokeasaurusrex Apologies, we were looking at the wrong organization when were verifying if release symbols were being uploaded or not.

We've confirmed that the correct release information is being sent to the correct organization in Sentry, and our issue is only the same as Petr reported, simply on macOS instead of just Windows.

We're going to update our CI script to allow this subprocess failure, though we'd ideally like to see a fix for this.

**Comment by szokeasaurusrex** [comment]
Okay @Mordil, glad to hear that everything is working as expected in Sentry, but sorry to hear that this problem is affecting you as well.

Would you be able to share a debug symbol file that reproduces this error (or the source code and steps to recreate such a debug file)? If your project is open source, you could just share the code/debug files that you noticed the problems with. Otherwise, if you have been able to create a minimal reproduction that you can share with me, it would greatly help me with debugging this problem.

**Comment by bjorn** [comment]
Since recently I'm seeing this issue on my Linux builds, though I'm not sure if it's just the same error message or whether the entire behavior is the same. I have the impression the entire upload fails in my case.

An example build running into this issue can be seen at https://github.com/mapeditor/tiled/actions/runs/9592646514/job/26451582147 (see the "Upload symbols and sources to Sentry" step):

```
Sucessfully installed sentry-cli 2.32.1
Done!
  INFO    2024-06-20 06:32:47.682623978 +00:00 sentry-cli was invoked with the following command line: "sentry-cli" "upload-dif" "--log-level=info" "--include-sources" "src" "."
  INFO    2024-06-20 06:32:47.682699229 +00:00 Issuing a command for Organization: mapeditor Project: tiled
error: Invalid checksum
```

The last succesfull build is at https://github.com/mapeditor/tiled/actions/runs/9562961181/job/26360475510.

The problem actually started happening after updating from Sentry 0.6.7 to Sentry 0.7.6. In both cases sentry-cli 2.32.1 was used.

**Comment by szokeasaurusrex** [comment]
@bjorn, this might be the same problem.

> The problem actually started happening after updating from Sentry 0.6.7 to Sentry 0.7.6

Which SDK version numbers are you referring to here?

**Comment by bjorn** [comment]
> Which SDK version numbers are you referring to here?

Whoops, that would be the version of the Sentry Native SDK at https://github.com/getsentry/sentry-native.

By downgrading the `sentry-native` version I've now confirmed that this uploading issue does not occur on 0.6.7 but does happen since the 0.7.0 version (this version made `crashpad` the the default, but I'm passing `-DSENTRY_BACKEND=breakpad`).

* Fails with 0.7.0: https://github.com/mapeditor/tiled/actions/runs/9643019019/job/26592093862 (never mind the green check, it's cause the step is marked `continue-on-error`)
* Works with 0.6.7: https://github.com/mapeditor/tiled/actions/runs/9643160290/job/26592537298

**Comment by szokeasaurusrex** [comment]
Okay, thank you for that information @bjorn, this is very helpful! One more question to ensure your issue is the same as what @petrmanek and @Mordil reported: despite the error message, are the debug files still showing up in Sentry?

**Comment by Mordil** [comment]
For the record, with additional testing, debug symbols were NOT being sent to Sentry and we identified the issue with something from one of our dependencies: https://github.com/filestack/filestack-ios/issues/80

**Comment by bjorn** [comment]
@szokeasaurusrex Yeah when this error occurs for me I don't see the debug files showing up in Sentry. I guess I'm not running into the same issue as @Mordil, since for me the only difference is whether I use Sentry Native 0.6.7 or Sentry Native 0.7.x.

For now I think I've been able to resolve this issue by only uploading the `.debug` files and not all `.o` files (see https://github.com/mapeditor/tiled/pull/3987, which changes from passing repository root to the the `AppDir` directory which contains only the installed files). I'm not sure if the `.o` files add any information.

**Comment by szokeasaurusrex** [comment]
@bjorn Are you still seeing full stack traces in Sentry after https://github.com/mapeditor/tiled/pull/3987? If yes, then probably you don't need the `.o` files you were uploading before

**Comment by bjorn** [comment]
> Are you still seeing full stack traces in Sentry after [mapeditor/tiled#3987](https://github.com/mapeditor/tiled/pull/3987)?

I will only know for sure once somebody manages to crash the Tiled 1.11 AppImage, but that's only a matter of time unfortunately. :-)

**Comment by szokeasaurusrex** [comment]
@petrmanek, @Mordil, @bjorn just wanted to check whether any of you are still experiencing this issue on the latest version of Sentry CLI, and/or whether you have identified any workarounds?

**Comment by isaacy13** [comment]
@szokeasaurusrex FWIW, I just updated to the latest version of sentry-cli (2.39.1) and still experiencing the same issue

- works as expected on local, but fails on Github pipelines
  - have been using it on local for months, successfully 
  - have never gotten it working on GH pipelines, has been a few hours since gotten pipeline setup

logs:
[gh-sentry-cli.txt](https://github.com/user-attachments/files/18134645/gh-sentry-cli.txt)

**Comment by isaacy13** [comment]
Validated my suspicion it's because GitHub runner is using different shell than my local

- GH runner using "C:\Program Files\PowerShell\7\pwsh.EXE"

Workaround for me was using cmd for that job

example:
```
- name: Upload debug symbols (debug mode)
        shell: cmd
        continue-on-error: true
        run: windows\sentry-cli.exe --log-level=debug --auth-token ${{ secrets.SENTRY_AUTH_TOKEN }} upload-dif -o themedeck -p themedeck-desktop-ffi .
```
- uses "C:\Windows\system32\cmd.EXE"

Might be worth a shot looking into if you're still running into this issue as a workaround
@petrmanek, @Mordil, @bjorn, @szokeasaurusrex

**Comment by petrmanek** [comment]
> Might be worth a shot looking into if you're still running into this issue as a workaround

My GitLab Runner also uses PowerShell but for legacy reasons our entire CI pipeline runs within the scope of `cmd.exe`, including Sentry.

**Comment by petrmanek** [comment]
> just wanted to check whether any of you are still experiencing this issue on the latest version of Sentry CLI, and/or whether you have identified any workarounds?

Yes, this is still an issue in our pipeline. Our current "workaround" has not changed since we [discussed it some time ago](https://github.com/getsentry/sentry-cli/issues/1906#issuecomment-1910091240); we simply allow the `sentry upload-dif` command to fail on Windows without crashing the entire pipeline. This is not something that I like to see in scripts, but it seems to work in acceptable manner.

**Comment by isaacy13** [comment]
Ran into this issue again -- was happening on local and pipeline

Fix was to call `flutter clean` before building, some funky files in build folder

**Comment by Mordil** [comment]
Yes, we're still running into this issue on the latest version of the Sentry CLI on a macOS runner using:

- macOS 14.7.1
- Sentry CLI 2.39.1

I've looked at the archives from our native iOS app when one of our dependencies is linked via Cocoapods vs SwiftPM to find differences, and there's nothing visible at the filesystem level.

The dependency in question is Filestack ([issue here](https://github.com/filestack/filestack-ios/issues/80)). When linked via SwiftPM we get the `[checksum ](error: Invalid checksum)` log

What is Sentry CLI even validating with that checksum?

**Comment by Mordil** [comment]
Looking deeper at older builds, we found that some dSYM files are no longer being created for dependencies that previously were.

Is this checksum comparing that there's no files to replace the existing ones?

**Comment by bjorn** [comment]
@isaacy13 I'm no longer seeing this issue after the workaround I had done at https://github.com/mapeditor/tiled/pull/3987, which actually I don't consider much of a workaround but should be better overall anyway.

@szokeasaurusrex To answer your question, yet, for Tiled 1.11, which was made after the above change, I still get source code references on my backtraces, so there is enough debug information.

**Comment by szokeasaurusrex** [comment]
@Mordil, please open a separate issue for [your problem](https://github.com/getsentry/sentry-cli/issues/1906#issuecomment-2546049351). This issue is specifically for uploading DIFs from Windows. The problem you are observing is on Mac, and thus out of scope for this issue

**Comment by szokeasaurusrex** [comment]
@petrmanek are you able to share a minimal reproduction with me? This would be very helpful as I try to debug this problem

**Comment by petrmanek** [comment]
> [@petrmanek](https://github.com/petrmanek) are you able to share a minimal reproduction with me? This would be very helpful as I try to debug this problem

What exactly did you have in mind? My current example is a proprietary C++/cmake project with a Windows CI pipeline that calls `sentry-cli upload-dif` in the final step.

I mean, I can create a whole new 'Hello World' C++/cmake repository with a Windows CI job and Sentry account, but that sounds a little bit like an overkill to me. Can you please specify what exactly would you consider helpful for troubleshooting this issue, besides the steps to reproduce and the log printout that I shared in my initial message?

**Comment by szokeasaurusrex** [comment]
@petrmanek I think it would be sufficient if you can share a non-proprietary debug file that, when uploaded with `sentry-cli upload-dif`, causes the error you are seeing.

I don't think I need an entire repo, I just would like to be able to run `sentry-cli` inside a debugger to see where the "Invalid checksum" error is coming from, since I currently have no idea where it is originating from, which makes it extremely difficult for me to figure out how I can fix the problem.

**Comment by szokeasaurusrex** [comment]
@petrmanek something else which would be helpful: If you are able to upgrade to Sentry CLI 2.40.0, try running the command with the `RUST_BACKTRACE` environment variable set to `1`, and provide the error logs you get with the environment variable set. That should hopefully give us some idea of where this error is coming from.

**Comment by isaacy13** [comment]
Ran into this issue again just now, `flutter clean` didn't help

Updated to Sentry CLI 2.40.0 and set RUST_BACKTRACE to 1

`\build\windows\x64\_deps\sentry-native-src\external\crashpad\third_party\zlib\zlib\google\test\data\Wrong CRC.zip` seems to be issue

Using `--no-zip` flag as workaround for now

Is there a way to ignore filepaths with sentry-cli (e.g.: .git)? Don't see anything in upload-dif help, may be missing it

CC @szokeasaurusrex 

```
  DEBUG   2025-01-20 11:50:43.340664500 -06:00 finished zip archive .\build\windows\x64\_deps\sentry-native-src\external\crashpad\third_party\zlib\zlib\google\test\data\test_posix_permissions.zip
  DEBUG   2025-01-20 11:50:43.340868600 -06:00 searching zip archive .\build\windows\x64\_deps\sentry-native-src\external\crashpad\third_party\zlib\zlib\google\test\data\Windows Special Names.zip
  DEBUG   2025-01-20 11:50:43.341409100 -06:00 trying to process First                                                      
  DEBUG   2025-01-20 11:50:43.342033200 -06:00 trying to process NUL                                                        
  DEBUG   2025-01-20 11:50:43.342671100 -06:00 trying to process NUL                                                        
  DEBUG   2025-01-20 11:50:43.343405600 -06:00 trying to process NUL.                                                       
  DEBUG   2025-01-20 11:50:43.344243300 -06:00 trying to process NUL .                                                      
  DEBUG   2025-01-20 11:50:43.344975700 -06:00 trying to process NUL.txt                                                    
  DEBUG   2025-01-20 11:50:43.345643 -06:00 trying to process NUL.tar.gz                                                    
  DEBUG   2025-01-20 11:50:43.346266100 -06:00 trying to process NUL..txt                                                   
  DEBUG   2025-01-20 11:50:43.346900200 -06:00 trying to process NUL...txt                                                  
  DEBUG   2025-01-20 11:50:43.347524400 -06:00 trying to process NUL .txt                                                   
  DEBUG   2025-01-20 11:50:43.348139800 -06:00 trying to process NUL  .txt                                                  
  DEBUG   2025-01-20 11:50:43.348750300 -06:00 trying to process NUL  ..txt                                                 
  DEBUG   2025-01-20 11:50:43.349522500 -06:00 trying to process Nul.txt
  DEBUG   2025-01-20 11:50:43.350411700 -06:00 trying to process nul.very long extension                                    
  DEBUG   2025-01-20 11:50:43.351102800 -06:00 trying to process  NUL.txt                                                   
  DEBUG   2025-01-20 11:50:43.351763300 -06:00 trying to process a/NUL                                                      
  DEBUG   2025-01-20 11:50:43.352513700 -06:00 trying to process CON                                                        
  DEBUG   2025-01-20 11:50:43.353540900 -06:00 trying to process PRN                                                        
  DEBUG   2025-01-20 11:50:43.354378100 -06:00 trying to process AUX                                                        
  DEBUG   2025-01-20 11:50:43.355198 -06:00 trying to process COM1                                                          
  DEBUG   2025-01-20 11:50:43.356161600 -06:00 trying to process COM2                                                       
  DEBUG   2025-01-20 11:50:43.356921600 -06:00 trying to process COM3                                                       
  DEBUG   2025-01-20 11:50:43.358067700 -06:00 trying to process COM4
  DEBUG   2025-01-20 11:50:43.358921600 -06:00 trying to process COM5                                                       
  DEBUG   2025-01-20 11:50:43.359883100 -06:00 trying to process COM6                                                       
  DEBUG   2025-01-20 11:50:43.360989900 -06:00 trying to process COM7                                                       
  DEBUG   2025-01-20 11:50:43.362084800 -06:00 trying to process COM8                                                       
  DEBUG   2025-01-20 11:50:43.363036500 -06:00 trying to process COM9                                                       
  DEBUG   2025-01-20 11:50:43.363862700 -06:00 trying to process LPT1                                                       
  DEBUG   2025-01-20 11:50:43.364841 -06:00 trying to process LPT2                                                          
  DEBUG   2025-01-20 11:50:43.365654100 -06:00 trying to process LPT3                                                       
  DEBUG   2025-01-20 11:50:43.366449300 -06:00 trying to process LPT4                                                       
  DEBUG   2025-01-20 11:50:43.367358 -06:00 trying to process LPT5
  DEBUG   2025-01-20 11:50:43.368086400 -06:00 trying to process LPT6                                                       
  DEBUG   2025-01-20 11:50:43.368783400 -06:00 trying to process LPT7                                                       
  DEBUG   2025-01-20 11:50:43.369452200 -06:00 trying to process LPT8                                                       
  DEBUG   2025-01-20 11:50:43.370135900 -06:00 trying to process LPT9                                                       
  DEBUG   2025-01-20 11:50:43.370747 -06:00 trying to process CLOCK$                                                        
  DEBUG   2025-01-20 11:50:43.371375100 -06:00 trying to process Last                                                       
  DEBUG   2025-01-20 11:50:43.371634400 -06:00 finished zip archive .\build\windows\x64\_deps\sentry-native-src\external\crashpad\third_party\zlib\zlib\google\test\data\Windows Special Names.zip
  DEBUG   2025-01-20 11:50:43.371950800 -06:00 searching zip archive .\build\windows\x64\_deps\sentry-native-src\external\crashpad\third_party\zlib\zlib\google\test\data\Wrong CRC.zip   
error: Invalid checksum                                       
  INFO    2025-01-20 11:50:43.385283100 -06:00 Skipping update nagger update check

 *  The terminal process "C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -Command .\windows\sentry-cli.exe --log-level=debug upload-dif -o themedeck -p themedeck-desktop-ffi ." terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it. 
```

**Comment by szokeasaurusrex** [comment]
@isaacy13 Yeah, based on the name of that file being `Wrong CRC.zip`, I assume that the ZIP file has an invalid CRC checksum, which would trigger this error. I don't think there is any way with the `debug-files upload` command to ignore certain paths (`upload-dif` is a deprecated alias to `debug-files upload`).

Are you calling Sentry CLI manually, or are you using a build plugin which internally calls Sentry CLI? If you are calling Sentry CLI manually, I always recommend passing the narrowest possible path to Sentry CLI where you know all of the debug files are stored

**Comment by isaacy13** [comment]
Gotcha thank you, I'm calling it manually, may play around with paths a bit

I set it to my root because wasn't sure if other necessary dlls hiding somewhere, path probably could be optimized

**Comment by szokeasaurusrex** [comment]
My suspicion here is that most likely, these failures are all being caused by the Sentry CLI attempting to search for debug files in a corrupted zip file (see my comment above: https://github.com/getsentry/sentry-cli/issues/1906#issuecomment-2604819900). I don't think anyone has shared a reproduction yet, but if someone has a reproduction which shows this error being caused by something other than a corrupted ZIP file, please share this with me – in that case, I will investigate further.

The solution will be to improve the error message to include the path to the corrupted ZIP file, and possibly make it a warning rather than an error which fails the upload. To workaround this problem, please ensure that there are no corrupted ZIP files anywhere in the path you are passing to Sentry CLI. I would recommend passing the narrowest path you can, which you know includes all of the debug files you need to upload, to avoid unintentionally including a corrupted ZIP file in the path.

I have opened #2405 to track the error message improvements. As it seems that #2405 is all we can do here, I am going to close this issue for now.
